doc/srv6/srv6_vpn.md

```diff
@@ -0,0 +1,287 @@
+
+We found that when sending 2M routes to orchagent, the orchagent will use about 3.5G of memory. Most of the memory is used by saimeta. the memory in saimeta is used for routing checks. In order to reduce memory usage and improving performance, we added a flag in the saimeta to decide whether to skip this routing check. By skipping this check, the memory usage will be reduced to 1.4G with 2M routes.
+
+## SRv6 Policy
+
+Segment Routing policy is defined in RFC 9256. SRv6 Policy's definition follows the same RFC. SRv6 Policy related information would be provided from FRR's pathd. Currently, we take a workaround approach which is described in "SRv6 policy download" section.
+
+The SRv6-policy information is only stored in the orchagent and won't be distributed to the ASIC\_DB. Therefore, when SRv6-policy are added and deleted, SAI entry will not be involved if the policy is not in use. SRv6 policy information would be consumed when a SRv6 policy route is added.
+
+The SRv6-policy content could be changed at any time. The changes could include segment adding/removing or weight updating. When a SRv6-policy content is updated, a backwalk is triggered to walk through all corresponding nexthop groups. Corresponding hardware information would be updated during the backwalk based on updated policy information. 
+
+![image](images/6Policy.png)
+
+# BFD Orch related Changes
+
+## SBFD
+
+In SRv6 deployment, we rely on SBFD (RFC 7880,  RFC7881) to protect SRv6 policy defined paths. Due to fast convergence and scale 's requirements, we would like to offload SBFD to hardware. The following changes are made for offloading SBFD sessions.
+
+1.  Add a new bfd encapsulation type enum SAI\_BFD\_ENCAPSULATION\_TYPE\_SRV6. When the SRv6 policy bfd session type is  SBFD Initiator  or SBFD echo, set SAI\_BFD\_SESSION\_ATTR\_BFD\_ENCAPSULATION\_TYPE attribute to SAI\_BFD\_ENCAPSULATION\_TYPE\_SRV6 if SRv6 is used for this SBFD session.
+    
+2.  Add a new  bfd session attr enum SAI\_BFD\_SESSION\_ATTR\_SRV6\_SIDLIST\_ID.  When the SRv6 policy bfd session type is  SBFD Initiator  or SBFD echo, we must set the sidlist id in SAI\_BFD\_SESSION\_ATTR\_SRV6\_SIDLIST\_ID to provide path information for SBFD SRv6 session.  The sidlist OID would be obtained via getBfdSidListOID in SRv6Orch.  
+    
+
+### BFD\_PEER schema
+
+There are three changes maded in BFD\_PEER schema.
+
+*   "local-discriminator" and "remote-discriminator" are added in BFD\_PEER schema. Different bfd sessions may have the same local IP address and remote IP address. We need use local and remote discriminator to differentiate these BFD sessions.
+    
+*   Three new BFD types are added, SBFD initiator , SBFD reflector and SBFD echo.
+    
+*   Two extra fields "endpoint" and "seglist-name" are added
+    
+
+The BFD\_PEER schema is shown as the following
+
+    community:
+    BFD_SESSION:{{vrf}}:{{ifname}}:{{ipaddr}}:{{local-discriminator}}
```

Comment:
Please create another table  `SBFD_SESSION `. With IPv6 ipaddr, it is challenging to parse if a key is present after the IP. So suggest have local-discriminator before the ipaddr.

Reply:
OK， I will update the new SBFD_SESSION schema.